tests/bls_tests.rs

```diff
@@ -0,0 +1,341 @@
+use ethereum_consensus::crypto::{
+    aggregate, aggregate_verify, fast_aggregate_verify, PublicKey, SecretKey, Signature,
+};
+use glob::glob;
+use hex;
+use serde::{Deserialize, Serialize};
+use serde_yaml;
+use std::fmt::Debug;
+use std::fs::File;
+
+fn decode_hex_with_prefix<T: AsRef<[u8]> + Debug>(data: T) -> Vec<u8> {
+    // a small check for empty strings
+    if data.as_ref().len() >= 2 {
+        hex::decode(&data.as_ref()[2..]).expect("is well-formed hex")
+    } else {
+        hex::decode(&data.as_ref()).expect("is well-formed hex")
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SigningInput {
+    privkey: String,
+    message: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SigningTestIO {
+    input: SigningInput,
+    output: String,
+}
+
+impl SigningTestIO {
+    fn verify_signature(&self) -> bool {
+        let input = &self.input;
+
+        let secret_key_bytes = decode_hex_with_prefix(&input.privkey);
+        let secret_key = match SecretKey::from_bytes(&secret_key_bytes) {
+            Ok(sk) => sk,
+            Err(_) => {
+                // this is the case empty secret key case
```

Comment:
which test case has this? i want to double-check the `~` isn't some yaml encoding

Reply:
It is this: [sign_case_zero_privkey](https://media.githubusercontent.com/media/ethereum/consensus-spec-tests/master/tests/general/phase0/bls/sign/small/sign_case_zero_privkey/data.yaml)